src/Drive.php

```diff
@@ -855,6 +855,42 @@ public function setPermissionExpiration(string $permissionId, ?\DateTimeImmutabl
         return $this->updatePermission($permissionId, $apiPermission);
     }
 
+    /**
+     * Search resource within a drive
+     *
+     * @return array<OcisResource>
+     * @throws SabreClientException
+     * @throws UnauthorizedException
+     * @throws TooEarlyException
+     * @throws ForbiddenException
+     * @throws InvalidResponseException
+     * @throws HttpException
+     * @throws \DOMException
+     * @throws SabreClientHttpException
+     * @throws BadRequestException
+     * @throws ConflictException
+     * @throws NotFoundException
+     * @throws InternalServerErrorException
+     */
+    public function searchResource(string $pattern, ?string $limit = null): array
+    {
+        $pattern .= " scope:" . $this->getId();
+        $webDavClient = $this->createWebDavClient();
+        $webDavClient->setCustomSetting($this->connectionConfig, $this->accessToken);
+
+        $responses = $webDavClient->sendReportRequest($pattern, $limit, '/remote.php/dav/spaces');
```

Comment:
why using `remote.php`? This is just historical route and should not be used in oCIS anymore

Reply:
need **remote.php** for api remote